library/src/main/java/com/hippo/unifile/TreeDocumentFile.java

```diff
@@ -233,13 +229,17 @@ public UniFile findFile(String displayName, boolean ignoreCase) {
             return null;
         }
 
-        final NamedUri[] result = DocumentsContractApi21.listFilesNamed(mContext, mUri);
-        for (NamedUri uri : result) {
-            if (Utils.equals(displayName, uri.name, ignoreCase)) {
-                return new TreeDocumentFile(this, mContext, uri.uri, displayName);
-            }
+        String documentId = DocumentsContract.getDocumentId(mUri);
+        documentId += "/" + displayName;
```

Comment:
See the comment here https://github.com/FooIbar/EhViewer/blob/bc7aa832b7a2ec3e7d262125ebe7ba54439c69b4/app/src/main/kotlin/com/hippo/unifile/DocumentsContractApi21.kt#L45

Reply:
I was rather wondering about that. I will adjust the code so that it checks the provider that is being used, and if it is not `ExternalStorageProvider` then it can fall back to the slower implementation that is more generic.

Reply:
After reconsidering, I think maybe we just leave a comment and assume it will work on most providers, in case some OEMs use different package names for `ExternalStorageProvider`.

Reply:
I've added https://github.com/tachiyomiorg/UniFile/pull/3 to document this, per your suggestion.